### PR TITLE
Do not attach and detach the EGL context on each frame in the rasterizer thread

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde

**Comment by chinmaygarde** [comment]
If we are not going to use it, maybe we can get rid of `GLContextClearCurrent` from the interface too. Either way, lgtm.

**Comment by j9brown** [comment]
Have you measured any difference in performance?

**Comment by jason-simmons** [comment]
The eglClearContext call was taking significant amounts of wall clock time in traces.

However, it's hard to gauge the full end-to-end effects.  Possibly the work that caused eglClearContext to block is merely being shifted elsewhere when the eglMakeCurrent/eglClearContext cycle is removed.

**Comment by jason-simmons** [comment]
I noticed some rendering oddities after screen rotations in the original patch.
This version has some more cleanup of the EGL surface/context lifecycle - PTAL

@chinmaygarde @abarth

**Comment by chinmaygarde** [comment]
lgtm. This looks cleaner that posting NotifyCreated on resize.

**Comment by abarth** [comment]
Is there a startup flash with this change?

File: shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -65,21 +66,27 @@ void PlatformViewAndroid::SurfaceCreated(JNIEnv* env,
   }
 
   ANativeWindow_release(window);
+
+  auto gl_surface = std::make_unique<GPUSurfaceGL>(surface_gl_.get());
+  NotifyCreated(std::move(gl_surface), [this, backgroundColor] {
+    rasterizer().Clear(backgroundColor, GetSize());
```
Comment (abarth) [inline shell/platform/android/platform_view_android.cc]: Does this `GetSize` call actually retrieve a reasonable size?  I thought I tested this before and got 0x0 at this point.

Comment (jason-simmons) [inline shell/platform/android/platform_view_android.cc]: In my tests GetSize() gets the proper window dimensions for the EGL surface by the time PlatformViewAndroid::SurfaceCreated is called.

The screen should be fully cleared to the background color at SurfaceCreated time with no flash.  Additionally, this should eliminate the flash during rotation where we had been clearing the surface midway through the window rotate animation.

**Comment by abarth** [comment]
Great!

**Comment by abarth** [comment]
LGTM

**Pull request merged**